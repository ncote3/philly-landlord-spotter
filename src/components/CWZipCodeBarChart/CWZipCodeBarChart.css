.cwZipCodeAnalysis {
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.chartSectionHeader {
    text-align: center;
    padding-top: 2vw;
    margin-bottom: 0 !important;
}

.dropdown-menu {
    height:200px;
    overflow-y:auto;
}
