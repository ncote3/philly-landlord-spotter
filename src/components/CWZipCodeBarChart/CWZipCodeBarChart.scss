@import './../philly-landlord-spotter/src/styles/colors';

.cwZipCodeAnalysis {
    margin-bottom: 2vw;
    background-color: $background_blur;
    box-shadow: 0 .35vw .35vw 0 rgba(0, 0, 0, 0.6);
    padding: 1.5vw 1.5vw .5vw;
    color: $white;
}

.chartSectionHeader {
    text-align: center;
    padding-top: 2vw;
    margin-bottom: 0 !important;
}

.dropdown-menu {
    height:200px;
    overflow-y:auto;
}
