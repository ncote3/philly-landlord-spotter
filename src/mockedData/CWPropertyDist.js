export const CWPropertyDist = [
    {"owner": "PHILADELPHIA HOUSING AUTH", "property_count": 4009},
    {"owner": "NEIGHBORHOOD RESTORATIONS", "property_count": 556},
    {"owner": "MARINE CLUB ASSOCIATES LP", "property_count": 319},
    {"owner": "REDEVELOPMENT AUTHORITY O", "property_count": 189},
    {"owner": "SIENA PLACE PLANNED COMMU", "property_count": 129},
    {"owner": "COMMONWEALTH OF PENNSYLVA", "property_count": 128},
    {"owner": "VENICE LOFTS PROPERTY OWN", "property_count": 128},
    {"owner": "PHILADELPHIA REDEVELOPMEN", "property_count": 119},
    {"owner": "JAMISON ROAD ASSOCIATES L", "property_count": 110},
    {"owner": "FEDERAL NATIONAL MORTGAGE", "property_count": 107},
    {"owner": "KELLEY INVESTMENT GROUP L", "property_count": 75},
    {"owner": "LSF9 MASTER PARTICIPATION", "property_count": 75},
    {"owner": "PHILADELPHIA LAND INVESTM", "property_count": 64},
    {"owner": "PHILADELPHIA LOTUS 01A LL", "property_count": 62},
    {"owner": "GINARK INVESTMENT AND MAN", "property_count": 61},
    {"owner": "PHILADELPHIA LOTUS 03A LL", "property_count": 60},
    {"owner": "PHILADELPHIA LOTUS 02A LL", "property_count": 56},
    {"owner": "EQUITY TRUST COMPANY CUST", "property_count": 56},
    {"owner": "BELMONT AFFORDABLE HOUSIN", "property_count": 56},
    {"owner": "PHILADELPHIA LOTUS 08 LLC", "property_count": 55},
    {"owner": "LWCA NEW YORK INVESTORS L", "property_count": 55},
    {"owner": "TRUSTEES OF THE UNIVERSIT", "property_count": 54},
    {"owner": "CITY BLOCK ACQUISITION VI", "property_count": 52}
]