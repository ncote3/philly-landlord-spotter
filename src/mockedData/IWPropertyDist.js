export const IWPropertyDist = [
  { owner: "PHILADELPHIA HOUSING AUTH", property_count: 343 },
  { owner: "CITY OF PHILA", property_count: 284 },
  { owner: "REDEVELOPMENT AUTHORITY", property_count: 190 },
  { owner: "PHILA HOUSING AUTHORITY", property_count: 134 },
  { owner: "TRS UNIV OF PENN", property_count: 102 },
  { owner: "PHILADELPHIA LAND BANK", property_count: 90 },
  { owner: "DREXEL UNIVERSITY", property_count: 82 },
  { owner: "PHILA REDEVELOPMENT AUTH", property_count: 65 },
  { owner: "BELMONT AFFORDABLE HOUSIN", property_count: 56 },
  { owner: "SIMON ASSOCIATES_LIMITED", property_count: 55 },
  { owner: "GEENA LLC", property_count: 47 },
  { owner: "TRUSTEES OF THE UNIVERSIT", property_count: 43 },
  { owner: "NEIGHBORHOOD RESTORATIONS", property_count: 42 },
  { owner: "CITY OF PHILADELPHIA", property_count: 39 },
  { owner: "SCHWARTZ SHELDON", property_count: 37 },
  { owner: "BROOKLYN HEIGHTS LP", property_count: 33 },
  { owner: "CAMPUS REALTY PROPERTIES", property_count: 32 },
  { owner: "STRONG PROPERTIES LLC", property_count: 31 },
  { owner: "WPRE II LP", property_count: 30 },
  { owner: "GILLER REALTY LLC", property_count: 26 },
  { owner: "AHM PROPERTIES LLC", property_count: 24 },
];
