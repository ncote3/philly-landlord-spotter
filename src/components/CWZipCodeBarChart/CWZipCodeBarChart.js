import React from "react";
import useAxios from "axios-hooks";
import Spinner from "react-bootstrap/Spinner";
import Alert from "react-bootstrap/Alert";
import BarChart from "../BarChart/BarChart";
import Container from "react-bootstrap/Container";
import './CWZipCodeBarChart.scss';

export default function CWZipCodeBarChart() {
    const [{ data, loading, error }] = useAxios(
        'https://www.landlord-spotter-backend.xyz/api/cw_property_dist/'
    );
    const zipCodeRes = {data, loading, error}

    if (zipCodeRes.loading) {
        return (
            <div data-testid='loading' style={{ marginTop: '10vh' }}>
                <Container className='individualZipCodeAnalysis' style={{padding: '2vw', textAlign: 'center'}}>
                    <Spinner animation="border" variant="light" size={'lg'} >
                        <span className="sr-only">Getting Zip Code Data...</span>
                    </Spinner>
                    <p>Getting Zip Code Data..</p>
                </Container>
            </div>
        )
    } else if (zipCodeRes.error) {
        return (
            <Container data-testid='error' className='individualZipCodeAnalysis' style={{padding: '2vw'}}>
                <Alert variant='danger'>Error getting zip code data!</Alert>
            </Container>
        )
    } else {
        return (
            <Container data-testid='data' className='cwZipCodeAnalysis'>
                <h1 className='chartSectionHeader'>Citywide Zip Code Analysis</h1>
                <Container>
                    <BarChart
                        zipData={zipCodeRes.data}
                        height={525}
                        width={900}
                    />
                </Container>
            </Container>
        )
    }
}
