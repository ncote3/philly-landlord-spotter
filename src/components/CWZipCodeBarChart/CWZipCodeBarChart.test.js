import React from "react";
import { unmountComponentAtNode } from "react-dom";
import { render as tlRender } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import CWZipCodeBarChart from "./CWZipCodeBarChart";
import pretty from "pretty";
import useAxios from "axios-hooks";
import { CWPropertyDist } from "../../mockedData/CWPropertyDist";

jest.mock("axios-hooks");

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("Axios Hook Aspects", () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it("renders correctly with a loading state", () => {
    useAxios.mockReturnValue([{ data: undefined, loading: true }]);
    const { queryByTestId } = tlRender(<CWZipCodeBarChart />, container);

    expect(queryByTestId("data")).toBeNull();
    expect(queryByTestId("loading")).not.toBeNull();
    expect(queryByTestId("error")).toBeNull();
    expect(container.innerHTML).toMatchSnapshot(pretty(container.innerHTML));
  });
  it("renders correctly with an error state", () => {
    useAxios.mockReturnValue([
      { data: undefined, loading: false, error: true },
    ]);
    const { queryByTestId } = tlRender(<CWZipCodeBarChart />, container);

    expect(queryByTestId("data")).toBeNull();
    expect(queryByTestId("loading")).toBeNull();
    expect(queryByTestId("error")).not.toBeNull();
    expect(container.innerHTML).toMatchSnapshot(pretty(container.innerHTML));
  });
  it("renders correctly with data", () => {
    useAxios.mockReturnValue([
      { data: CWPropertyDist, loading: false, error: false },
    ]);
    const { queryByTestId } = tlRender(<CWZipCodeBarChart />, container);

    expect(queryByTestId("data")).not.toBeNull();
    expect(queryByTestId("loading")).toBeNull();
    expect(queryByTestId("error")).toBeNull();
    expect(container.innerHTML).toMatchSnapshot(pretty(container.innerHTML));
  });
});
