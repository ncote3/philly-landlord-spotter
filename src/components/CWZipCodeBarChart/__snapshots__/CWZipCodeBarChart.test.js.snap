// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Axios Hook Aspects renders correctly with a loading state 1`] = `""`;

exports[`Axios Hook Aspects renders correctly with an error state 1`] = `""`;

exports[`Axios Hook Aspects renders correctly with data 1`] = `""`;
